### PR TITLE
438 formula 513 from nen en 1993 52008

### LINKED ISSUE #438: Formula 5.13 from NEN-EN 1993-5:2008

### PR DESCRIPTION
**Pull request by SZeltaat**
## Description
Implementation of formula 5.13 of NEN-EN 1993-5:2008

Fixes #438 

## Type of change
- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

## Checklist:

- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: blueprints/codes/eurocode/nen_en_1993_5_2008/chapter_5_ultimate_limit_states/formula_5_13.py [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+        gamma_m0: DIMENSIONLESS,
+        gamma_m1: DIMENSIONLESS,
+        chi: DIMENSIONLESS,
+        m_c_rd: KNM,
+    ) -> None:
+        r"""
+        Simplified buckling check.
+
+        Provided that the boundary conditions are supplied by elements (anchor, earth support, capping beam, etc.)
+        that give positional restraint corresponding to the non-sway buckling mode, this check may be used:
```
Comment (GerjanDorgelo) [inline blueprints/codes/eurocode/nen_en_1993_5_2008/chapter_5_ultimate_limit_states/formula_5_13.py:32]: can you add that its only valid for class 1, 2 and 3?

File: tests/codes/eurocode/nen_en_1993_5_2008/chapter_5_ultimate_limit_states/test_formula_5_13.py [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+
+    @pytest.mark.parametrize(
+        ("n_ed", "m_ed", "a", "f_y", "gamma_m0", "gamma_m1", "chi", "m_c_rd"),
+        [
+            (-100.0, 50.0, 2000.0, 355.0, 1.0, 1.0, 0.9, 100.0),  # n_ed is negative
+            (100.0, -50.0, 2000.0, 355.0, 1.0, 1.0, 0.9, 100.0),  # m_ed is negative
+            (100.0, 50.0, -2000.0, 355.0, 1.0, 1.0, 0.9, 100.0),  # a is negative
+            (100.0, 50.0, 2000.0, -355.0, 1.0, 1.0, 0.9, 100.0),  # f_y is negative
+            (100.0, 50.0, 2000.0, 355.0, 0.0, 1.0, 0.9, 100.0),  # gamma_m0 is zero
+            (100.0, 50.0, 2000.0, 355.0, 1.0, 0.0, 0.9, 100.0),  # gamma_m1 is zero
```
Comment (GerjanDorgelo) [inline tests/codes/eurocode/nen_en_1993_5_2008/chapter_5_ultimate_limit_states/test_formula_5_13.py:40]: can you add tests for all others that cannot be zero?

File: blueprints/codes/eurocode/nen_en_1993_5_2008/chapter_5_ultimate_limit_states/formula_5_13.py (hunk 1 above)
Comment (SZeltaat) [reply blueprints/codes/eurocode/nen_en_1993_5_2008/chapter_5_ultimate_limit_states/formula_5_13.py:32]: Oh! Of course! Done!

File: blueprints/codes/eurocode/nen_en_1993_5_2008/chapter_5_ultimate_limit_states/formula_5_13.py [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+        f_y : MPA
+            [$MPa$] Yield strength of the material [$MPa$].
+        gamma_m0 : DIMENSIONLESS
+            [$\gamma_M0$] Partial factor according to 5.1.1 (4) [-].
+        gamma_m1 : DIMENSIONLESS
+            [$\gamma_M1$] Partial factor according to 5.1.1 (4) [-].
+        chi : DIMENSIONLESS
+            [$\chi$] Buckling coefficient from 6.3.1.2 of EN 1993-1-1 [-].
+        m_c_rd : KNM
+            [$kNm$] Design moment resistance of the cross-section [$kNm$].
```
[the remark this answers is not in the document]
Comment (SZeltaat) [reply blueprints/codes/eurocode/nen_en_1993_5_2008/chapter_5_ultimate_limit_states/formula_5_13.py]: 5.2.2 (2) Done!

File: tests/codes/eurocode/nen_en_1993_5_2008/chapter_5_ultimate_limit_states/test_formula_5_13.py (hunk 2 above)
Comment (SZeltaat) [reply tests/codes/eurocode/nen_en_1993_5_2008/chapter_5_ultimate_limit_states/test_formula_5_13.py:44]: Sharp! Sure! Done.

File: blueprints/codes/eurocode/nen_en_1993_5_2008/chapter_5_ultimate_limit_states/formula_5_13.py [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+        )
+        raise_if_less_or_equal_to_zero(
+            a=a,
+            f_y=f_y,
+            chi=chi,
+            m_c_rd=m_c_rd,
+            gamma_m0=gamma_m0,
+            gamma_m1=gamma_m1,
+        )
+        n_pl_rd: KN = a * f_y / gamma_m0 * N_TO_KN
```
[the remark this answers is not in the document]
Comment (SZeltaat) [reply blueprints/codes/eurocode/nen_en_1993_5_2008/chapter_5_ultimate_limit_states/formula_5_13.py:88]: Good point! I'm going with your suggestion, because I find it hard to work that in the formula otherwise.

**Review by SZeltaat — COMMENTED** [review]
Thank you for your sharp review Gerjan!
I've made the changes! 😉

File: blueprints/codes/eurocode/nen_en_1993_5_2008/chapter_5_ultimate_limit_states/formula_5_13.py [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+        return (n_ed / (chi * n_pl_rd * (gamma_m0 / gamma_m1)) + 1.15 * m_ed / (m_c_rd * (gamma_m0 / gamma_m1))) <= 1.0
+
+    def latex(self) -> LatexFormula:
+        """Returns LatexFormula object for formula 5.13."""
+        _equation: str = (
+            r"\frac{N_{Ed}}{\chi \cdot (A \cdot f_{y} / \gamma_{M0}) \cdot \left( \frac{\gamma_{M0}}{\gamma_{M1}} \right)} + "
```
Comment (GerjanDorgelo) [inline blueprints/codes/eurocode/nen_en_1993_5_2008/chapter_5_ultimate_limit_states/formula_5_13.py:94]: the N_TO_KN factor 1000 gets lost visually. Could you add the "/1000"?

**Review by GerjanDorgelo — COMMENTED** [review]
Final detail before merge :)

File: blueprints/codes/eurocode/nen_en_1993_5_2008/chapter_5_ultimate_limit_states/formula_5_13.py (hunk 5 above)
Comment (SZeltaat) [reply blueprints/codes/eurocode/nen_en_1993_5_2008/chapter_5_ultimate_limit_states/formula_5_13.py:94]: Good point!
I've made the changes in the numerical form of A and f_y:

```python
                "A": f"{self.a:.2f} / 10^6",
                "f_{y}": f"{self.f_y:.1f} \\cdot 10^3",
```

image

**Review by SZeltaat — COMMENTED** [review]
Thanks for your sharp comment! I've made the changes. Could you please check if this PR is ready to be approved?

**Comment by GerjanDorgelo** [comment]
Love it!

**Review by GerjanDorgelo — APPROVED** [review]
